File: 2013-04-19_simple_pub_sub_js/subscribe.js
var redis = require('redis');
var client = redis.createClient();

process.argv.slice(2).forEach(function (channel, i) {

    client.subscribe(channel, function () {
        console.log('Subscribing to ' + channel + ' channel');
    });
});

client.on('message', function (channel, msg) {
    console.log("%s: %s", channel, msg);
});

File: 2013-04-19_simple_pub_sub_js/subscribe.rb
require 'rubygems'
require 'redis'

l = ARGV.length-1
channels = ARGV[0..l]
puts channels

$redis = Redis.new(:timeout => 0)
$redis.subscribe(channels.to_a) do |on|
  on.message do |channel, msg|
    puts "#{channel}: #{msg}"
  end
end
